crates/bevy_math/src/cubic_splines.rs

```diff
@@ -306,50 +321,64 @@ impl<P: Point> CubicGenerator<P> for CubicBSpline<P> {
         expected: usize,
-        /// Provided knot vector length
+        /// Provided number of knots
         provided: usize,
     },
-    /// Knot vector has invalid values. Values of a knot vector must be nondescending, meaning the
-    /// next element must be greater than or equal to the previous one.
-    #[error("Invalid knot vector values: elements are not nondescending")]
-    InvalidKnotVectorValues,
-    /// Provided weights vector didn't have the same amount of values as the control points vector.
-    #[error("Invalid weights vector length: expected {expected}, provided {provided}")]
-    WeightsVectorMismatch {
-        /// Expected weights vector length
+    /// The provided knots had a descending knot pair. Subsequent knots must
+    /// either increase or stay the same.
+    #[error("Invalid knots: contains descending knot pair")]
+    DescendingKnots,
+    /// The provided knots were all equal. Knots must contain at-least one increasing pair.
+    #[error("Invalid knots: all knots are equal")]
+    ConstantKnots,
+    /// Provided a different number of weights and control points.
+    #[error("Incorect number of weights: expected {expected}, provided {provided}")]
+    WeightsMismatch {
+        /// Expected number of weights
         expected: usize,
-        /// Provided weights vector length
+        /// Provided number of weights
         provided: usize,
     },
-    /// The amount of control points provided is less than 4.
+    /// The number of control points provided is less than 4.
     #[error("Not enough control points, at least 4 are required, {provided} were provided")]
     NotEnoughControlPoints {
-        /// The amount of control points provided
+        /// The number of control points provided
         provided: usize,
     },
 }
 
-/// A cubic non-uniform rational B-spline (NURBS). Generates a smooth curve from a
-/// sequence of control points by interpolating between four points at a time.
+/// Non-uniform Rational B-Splines (NURBS) are powerful generalization of the [`CubicBSpline`] which can
+/// represent a much more diverse class of curves (like perfect circles and ellipses).
+///
+/// ### Non-uniformity
+/// The 'NU' part of NURBS stands for "Non-Uniform". This has to do with a parameter called 'knots'.
+/// The knots are a non-decreasing sequence of floating point numbers. The first and last three pairs of
+/// knots controll the behavior of the curve as it approaches it's endpoints. The intermediate pairs
+/// each control the length of one segment of the curve. Multiple repeated knot values are called
+/// "knot multiplicity". Knot multiplicity in the intermediate knots causes a "zero-length" segments,
+/// and can create sharp corners.
+///
+/// ### Rationality
+/// The 'R' part of NURBS stands for "Rational". This has to do with NURBS allowing each control point to
+/// be assigned a weighting, which controls how much it effects the curve recitative to the other points.
```

Comment:
```suggestion
/// ### Rationality
/// The 'R' part of NURBS stands for "Rational". This has to do with NURBS allowing each control point to
/// be assigned a weighting, which controls how much it effects the curve compared to the other points.
```

"Recitative" is a quite obscure word, so I think replacing it with something more common would make the documentation clearer for more people.